packages/connector-tencent-email/src/index.ts

```diff
@@ -0,0 +1,148 @@
+import type {
+  CreateConnector,
+  EmailConnector,
+  GetConnectorConfig,
+  SendMessageFunction,
+} from '@logto/connector-kit';
+import {
+  ConnectorError,
+  ConnectorErrorCodes,
+  ConnectorType,
+  validateConfig,
+} from '@logto/connector-kit';
+import { assert } from '@silverhand/essentials';
+import { HTTPError } from 'got';
+
+import { isErrorResponse, request } from '@/http';
+import type { MailTemplateType, SendTencentMailConfig, ValueNames } from '@/types';
+import { sendTencentMailConfigGuard } from '@/types';
+
+import { defaultMetadata } from './constant';
+
+function getTemplate(templates: MailTemplateType[], usageType: string): MailTemplateType {
+  const template = templates.find((template) => template.usageType === usageType);
+  assert(
+    template,
+    new ConnectorError(
+      ConnectorErrorCodes.TemplateNotFound,
+      `Template not found for type: ${usageType}`
+    )
+  );
+
+  return template;
+}
+
+function buildFromName(fromName: string, fromAddress: string): string {
+  return fromName.length > 0 ? `${fromName} <${fromAddress}>` : fromAddress;
+}
+
+function buildTemplatePayload(
+  template: MailTemplateType,
+  templateParameters: Record<typeof ValueNames[number], string>
+): Record<string, string> {
+  return template.params.reduce<Record<string, string>>((accumulator, parameter) => {
+    // eslint-disable-next-line @silverhand/fp/no-mutation
+    accumulator[parameter.name] = templateParameters[parameter.value];
+
+    return accumulator;
+  }, {});
+}
+
+function transformString(...values: Array<string | undefined>): string {
+  return values.find((item) => item && item.length > 0) ?? '';
+}
```

Comment:
What does this function do? Should we use `.map()`?

Reply:
There may be a better way to just use it to get the first valid string in an array.